Improve404

Andtalath opened the pull request.

stakater-user commented:
@rasheedamir Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/StakaterCloudDocs/commit/2f77854886d65b51829f604180b0bd4756c142e4/checks) has Failed!

stakater-user commented:
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-147-8007e124`

rasheedamir commented:
we might have customer use cases where we need to support external-dns; so, no need to delete it

Andtalath committed:
improve handling of error pages

the current version can handle files missing, but it can not handle
directories missing.

https://docs.stakater.com/mto/main/reference-guides/admin-clusterrole.html
Fails to show a nice looking 404.

https://docs.stakater.com/mto/main/obviously_missing_file
succeeds.

With this change, both should work.

Not sure what internal does or if is needed, have tested on a local
nginx only.

Andtalath committed:
needed for 403

I realized try_files failed to handle 403, so this repairs that part

Andtalath committed:
Merge branch 'main' into improve404

karl-johan-grahn commented on default.conf:
```diff
@@ -2,6 +2,13 @@ server {
+    location = /404.html {
+        internal;
+    }
+    location / {
```
duplicated, already on line 12

Andtalath replied:
Fixed, I got confused when trying to fix the merge conflict I didn't understand.

Andtalath committed:
remove duplicate field
